From 5cb269a74ace5da67a9065ae1520c5420858bf9d Mon Sep 17 00:00:00 2001
From: Luca Boccassi <luca.boccassi@gmail.com>
Date: Fri, 17 Oct 2025 15:39:09 +0100
Subject: [PATCH] ci: add bpftool workaround to codeql job too

(cherry picked from commit e9fd2bbfffc5c2c7cd1ea0a288d5435fc15e387f)

Related: RHEL-155394
---
 .github/workflows/codeql.yml | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/.github/workflows/codeql.yml b/.github/workflows/codeql.yml
index 51034783ca..443221fa11 100644
--- a/.github/workflows/codeql.yml
+++ b/.github/workflows/codeql.yml
@@ -44,7 +44,14 @@ jobs:
         languages: ${{ matrix.language }}
         config-file: ./.github/codeql-config.yml
 
-    - run: sudo -E .github/workflows/unit_tests.sh SETUP
+    - run: |
+        sudo -E .github/workflows/unit_tests.sh SETUP
+        # TODO: drop after we switch to ubuntu 26.04
+        bpftool_binary=$(find /usr/lib/linux-tools/ /usr/lib/linux-tools-* -name 'bpftool' -perm /u=x 2>/dev/null | sort -r | head -n1)
+        if [ -n "$bpftool_binary" ]; then
+          sudo rm -f /usr/bin/bpftool
+          sudo ln -s "$bpftool_binary" /usr/bin/
+        fi
 
     - name: Autobuild
       uses: github/codeql-action/autobuild@f09c1c0a94de965c15400f5634aa42fac8fb8f88
